compiler/src/dotty/tools/dotc/typer/Inferencing.scala

```diff
@@ -412,7 +412,7 @@ trait Inferencing { this: Typer =>
     // `qualifying`.
 
     val ownedVars = state.ownedVars
-    if (ownedVars.size > 0) {
+    if (ownedVars.size > locked.size) {
```

Comment:
This seems to be a miracle -- how this change could have such a big impact on performance.